### PR TITLE
S4531382 ADNI GFNet Model

### PR DESCRIPTION
**Pull request by Mhope7**
Finalized GFNet including data loader, model, training script, testing script and README.md.

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm solves the problem appropriately. 
- Accuracy in testing dataset: 0.8022. 

**Model and functions:**

- It correctly uses PyTorch to construct the GFNet models and functions.
- With data augmentation.

**Code design:** Good.

**Code comment and docstring:**

- Good code comments 
- Good function docstrings

**Difficulty:** Hard.

**Additional Comments:**

- Minimal commits, just some copy-and-paste code.
- The module order in the report is not appropriate, for example, the data preprocessing and repo structure part are listed after the training/testing result. Please note the format of the academic report.
-  A conclusion is expected in the report.
- Training and validation accuracy plot is expected to be shown in the report.

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request shows no incorporation of previous feedback .
NO description added in the Pull Request . 

File Organizing: Well-organized files.

Commit Log:
Commit messages are not  meaningful .
Commits are not  regularly made
Documentation:
The README file provides a clear explanation of the algorithm and the problem it solves.
Code comments and docstrings are included .
Proper GitHub markdown formatting is used without organized headings, lists, and code blocks.
Can add about Conclusion of the report 
Should Discuss more about testing training results .

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Good solution to problem. </td><td></td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. Uses Data Aug.</td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard.  GFNet (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. Minimal.</td><td>-1</td></tr>
	<tr><td></td><td>Some/Adequate Progressive commits. Lacks flow.</td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. Poor stucture.</td><td>-2</td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Acceptable. </td><td>-1</td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is adequate. Minimal. Give a  high level intro to the approach.</td><td>-1</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-7</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by hanemma7moud** [comment]
Mackenzie Hope
